ci: OpenSUSE Leap 15.6 distro build fails on libncurses6 / devel_basis dependency conflict

## Symptom

The **OpenSUSE Leap 15.6** job in the `CMake Distribution Native Builds` workflow (`.github/workflows/cmake_distros.yml`) fails for every PR and for `development` itself. Other distros in the same workflow (Tumbleweed, Alpine Edge, Arch, Fedora, Debian Sid, Mint 22) pass.

Confirmed failing on:
- `development` run #511 (2026-04-28, after #2927 merge) — https://github.com/gridcoin-community/Gridcoin-Research/actions/runs/25146839614
- PRs #2923 and #2924 runs (just observed)

## Root cause

The failure is in `install_dependencies.sh` step 1, before any of our code is touched. `zypper install -y -t pattern devel_basis` hits an unresolvable dependency conflict:

```
Problem: 1: the to be installed pattern:devel_basis-20170319-lp156.3.2.x86_64
         requires 'patterns-devel-base-devel_basis', but this requirement
         cannot be provided
not installable providers: patterns-devel-base-devel_basis-20170319-lp156.3.2.x86_64[repo-oss]

 Solution 1: downgrade of libncurses6-6.1-150000.5.33.1.x86_64 to libncurses6-6.1-150000.5.30.1.x86_64
 Solution 2: do not install pattern:devel_basis-20170319-lp156.3.2.x86_64
 Solution 3: break pattern:devel_basis-20170319-lp156.3.2.x86_64 by ignoring some of its dependencies

Choose from above solutions by number or cancel [1/2/3/c/d/?] (c): c
##[error]Process completed with exit code 4.
```

`patterns-devel-base-devel_basis-20170319-lp156.3.2` was built against `libncurses6-6.1-150000.5.30.1`. The current Leap 15.6 update repo ships `libncurses6-6.1-150000.5.33.1`, which is incompatible. With `-y` the installer answers each per-package prompt with "yes," but it does not pick a solution number for the conflict resolver — it falls through to the default `c` (cancel) and zypper exits 4.

Same root cause affects every fresh `opensuse/leap:15.6` container, so every workflow run is broken until either the upstream pattern is rebuilt against the newer libncurses6 or the workflow tells zypper which solution to take.

## Where the failure happens

- Workflow: `.github/workflows/cmake_distros.yml` lines 30-33 (matrix entry for OpenSUSE Leap 15.6, `prep_cmd` is `zypper refresh && zypper install -y sudo which git`).
- Script: `install_dependencies.sh:225` — `sudo zypper install -y -t pattern devel_basis`.

## Suggested fixes (any one of these is enough)

1. **Pass `--force-resolution` and a solver argument to zypper** so it picks solution 1 (downgrade libncurses6) automatically:
   ```sh
   sudo zypper --non-interactive install --force-resolution -y -t pattern devel_basis
   ```
   Or the equivalent answer-file approach with `--solver-focus Update` plus `--allow-downgrade`.

2. **Switch the matrix entry to a Leap 15.6 update tag that ships the rebuilt pattern** (e.g., once openSUSE pushes a `patterns-devel-base-devel_basis` rebuilt against libncurses6 5.33.1).

3. **Drop the `devel_basis` pattern install** in the openSUSE branch of `install_dependencies.sh` and replace it with an explicit list of build packages (gcc, make, autoconf, automake, libtool, pkg-config, …). This is what we already do implicitly elsewhere; it removes the upstream-pattern coupling.

Option 1 is the smallest patch and gets CI green immediately. Option 3 is the most robust long-term fix but is a bigger script change.

## Reproduction

Run `CMake Distribution Native Builds` on any branch — only the OpenSUSE Leap 15.6 matrix entry will fail; all other distros pass. The conflict is reproducible locally with `docker run --rm -it opensuse/leap:15.6 sh -c 'zypper -n refresh && zypper -n install -y sudo which git && zypper -n install -y -t pattern devel_basis; echo exit=$?'` (will exit 4).

## Notes

- This is **not** caused by any in-flight PR — it predates current PRs #2923 and #2924 and reproduces on `development`.
- Open for grabs as a `good first issue`-adjacent CI fix; tagging `help wanted`.

--- a/.github/workflows/cmake_distros.yml
+++ b/.github/workflows/cmake_distros.yml
@@ -27,8 +27,8 @@ jobs:
             prep_cmd: "dnf install -y which sudo git"
             build_args: ""
 
-          - distro: OpenSUSE Leap 15.6
-            image: opensuse/leap:15.6
+          - distro: OpenSUSE Leap 16.0
+            image: opensuse/leap:16.0
             prep_cmd: "zypper refresh && zypper install -y sudo which git"
             build_args: "CC=gcc-13 CXX=g++-13"
 

